jerry-core/ecma/builtin-objects/ecma-builtin-helpers-sort.c

```diff
@@ -93,7 +101,7 @@ ecma_builtin_helper_array_to_heap (ecma_value_t *array_p, /**< heap data array *
   JERRY_ASSERT (parent <= right);
   array_p[parent] = swap;
 
-  if (ecma_is_value_empty (ret_value))
+  if (!ECMA_IS_VALUE_ERROR (ret_value))
```

Comment:
you can return `ret_value` directly. Please ensure that the function may return with empty or error value.